### PR TITLE
Ensure compatibility with Twig 3.9

### LINKED ISSUE #7121: Can't access backend after upgrade to 4.13.41
### Affected version(s)

4.13.41

### Description

After upgrading one of my Contao installations i can not access the backend.
An "internal server error" is shown:

`An exception has been thrown during the rendering of a template ("Call to undefined function Contao\CoreBundle\Twig\Extension\twig_escape_filter_is_safe()").`

### PR DESCRIPTION
**Pull request by leofeyer**
Fixes #7121

### CODE REVIEW

**Comment by m-vo** [comment]
Probably related: https://github.com/twigphp/Twig/issues/4022

I think we'll need to debug the contents of one of the compiled templates to see what's going on. I guess either the compiled output of the `ModuleNode` or the `Template` have changed and don't contain the variable anymore for our `PhpTemplateProxyNode` or something wraps our code (think of an anonymous function without binding variables).

**Comment by ausi** [comment]
See https://github.com/leofeyer/contao/pull/12 09c6f4b1f2d10ef4c1c53918451aac79b4b077ae

the solution is similar to what Twig does here: https://github.com/twigphp/Twig/blob/2d169bdca2be810e71fef2df2279aa02771fab33/src/Extension/CoreExtension.php#L1848

File: core-bundle/tests/Twig/Extension/ContaoExtensionTest.php [hunk 1]
```diff
@@ -377,25 +377,33 @@ public static function provideTemplateNames(): iterable
...
     }
 
     public static function provideTwigFunctionSignatures(): iterable
     {
+        // Make sure the functions outside the class scope are loaded
+        new \ReflectionClass(EscaperExtension::class);
+
+        // Forward compatibility with twig/twig 4
+        if (method_exists(EscaperExtension::class, 'escape')) {
+            $escape = new \ReflectionMethod(EscaperExtension::class.'::escape');
```
Comment (ausi) [inline core-bundle/tests/Twig/Extension/ContaoExtensionTest.php]: This usage seems to be deprecated in PHP 8.4.

```suggestion
            $escape = new \ReflectionMethod(EscaperExtension::class, 'escape');
```

File: core-bundle/tests/Twig/Extension/ContaoExtensionTest.php [hunk 2]
```diff
@@ -405,12 +413,14 @@ public static function provideTwigFunctionSignatures(): iterable
...
-            'twig_escape_filter_is_safe',
-            [
-                [Node::class, 'filterArgs'],
-            ],
-        ];
+        // Backwards compatibility with twig/twig <3.9
+        if (\function_exists('twig_escape_filter_is_safe')) {
+            $escapeIsSafe = new \ReflectionFunction('twig_escape_filter_is_safe');
+        } else {
+            $escapeIsSafe = new \ReflectionMethod(EscaperExtension::class.'::escapeFilterIsSafe');
```
Comment (ausi) [inline core-bundle/tests/Twig/Extension/ContaoExtensionTest.php]:
```suggestion
            $escapeIsSafe = new \ReflectionMethod(EscaperExtension::class, 'escapeFilterIsSafe');
```

**Review by ausi — APPROVED** [review]

**Pull request merged**